src/ui/public/share/directives/share.js

```diff
@@ -1,18 +1,146 @@
+import {
+  getUnhashableStatesProvider,
+  unhashUrl,
+} from 'ui/state_management/state_hashing';
+import Notifier from 'ui/notify/notifier';
+
+import urlShortenerProvider from '../lib/url_shortener';
+
 import uiModules from 'ui/modules';
 import shareTemplate from 'ui/share/views/share.html';
 const app = uiModules.get('kibana');
 
-app.directive('share', function () {
+app.directive('share', function (Private) {
+  const getUnhashableStates = Private(getUnhashableStatesProvider);
+  const urlShortener = Private(urlShortenerProvider);
+
   return {
     restrict: 'E',
     scope: {
       objectType: '@',
       objectId: '@',
-      setAllowEmbed: '&?allowEmbed'
+      allowEmbed: '@',
     },
     template: shareTemplate,
-    controller: function ($scope) {
-      $scope.allowEmbed = $scope.setAllowEmbed ? $scope.setAllowEmbed() : true;
+    controllerAs: 'share',
+    controller: function ($scope, $location, globalState) {
+      if ($scope.allowEmbed !== 'false' && $scope.allowEmbed !== undefined) {
+        throw new Error('allowEmbed must be "false" or undefined');
+      }
+
+      // Default to allowing an embedded IFRAME, unless it's explicitly set to false.
+      this.allowEmbed = $scope.allowEmbed === 'false' ? false : true;
```

Comment:
I tried the switch idea, but couldn't figure out a case to match the `$scope.allowEmbed !== undefined` condition.

Re the original suggestion, I think equality comparisons are easier to think about than inequality, especially when comparing booleans. When you see an equality comparison in a ternary, you can associate the happy path with the values and variables in the comparison. But with an inequality comparison, you need to think of the opposite of the compared value to imagine the happy path, which I think is one extra unnecessary step.

Is the ternary the problem? Do you think this is easier to follow @epixa ?

```javascript
this.allowEmbed = true;

// Default to allowing an embedded IFRAME, unless it's explicitly set to false.
if ($scope.allowEmbed === 'false') {
  this.allowEmbed = false;
}
```

I can also make the comment clearer if you have any suggestions.